src/main/java/com/google/devtools/build/lib/actions/ResourceManager.java

```diff
@@ -392,6 +406,21 @@ private boolean areResourcesAvailable(ResourceSet resources) {
     double availableIo = availableResources.getIoUsage();
     int availableLocalTestCount = availableResources.getLocalTestCount();
 
+    double remainingRam = availableRam - usedRam;
```

Comment:
How about putting the assignment into the else block?

Reply:
I'm unsure what you're thinking  

As it stands, `remainingRam` is assigned here so that we fall back to identical behavior if either (a) the experimental flag is off or (b) there is an exception when trying to retrieve the currently used memory.  If you look, the check at the bottom has been rephrased to now be in terms of `remainingRam`.